services/viva-ms/lambdas/createVivaCase.js

```
@@ -152,3 +158,70 @@ async function putRecurringVivaCase(PK, workflowId, period) {
 
   return caseItem;
 }
+
+async function getUser(PK) {
+  const personalNumber = PK.replace('USER#', '');
+  const params = {
+    TableName: config.users.tableName,
+    Key: {
+      personalNumber,
+    },
+  };
+
+  const [error, dbResponse] = await to(dynamoDB.call('get', params));
+  if (!dbResponse) {
+    console.error('(cases-api) DynamoDb query on users table failed', error);
+    return;
+  }
+
+  return dbResponse.Item;
+}
+
+async function getFormTemplates(forms) {
+  const formTemplates = {};
+  for (const key of Object.keys(forms)) {
+    const params = {
+      TableName: config.forms.tableName,
+      Key: {
+        PK: `FORM#${key}`,
+      },
+    };
+    const [error, dbResponse] = await to(dynamoDB.call('get', params));
+
+    if (!dbResponse) {
+      console.error('(cases-api) DynamoDb query on forms table failed', error);
+      continue;
+    }
+    formTemplates[key] = dbResponse.Item;
+  }
+
+  return formTemplates;
+}
+
+async function getLastUpdatedCase(PK, provider) {
+  const params = {
+    TableName: config.cases.tableName,
+    KeyConditionExpression: 'PK = :pk',
+    FilterExpression: 'begins_with(#status.#type, :statusTypeClosed) and #provider = :provider',
+    ExpressionAttributeNames: {
+      '#status': 'status',
+      '#type': 'type',
+      '#provider': 'provider',
+    },
+    ExpressionAttributeValues: {
+      ':pk': PK,
+      ':statusTypeClosed': 'closed',
+      ':provider': provider,
+    },
+  };
+
+  const [error, dbResponse] = await to(dynamoDB.call('query', params));
+  if (!dbResponse) {
+    console.error('(cases-api) DynamoDb query on cases table failed', error);
+    return;
+  }
```

Throw error instead of return.